core/src/main/resources/hudson/model/AllView/noJob.groovy

```diff
@@ -4,40 +4,142 @@ import hudson.model.Computer
 import hudson.model.Item
 import jenkins.model.Jenkins
 
+def l = namespace(lib.LayoutTagLib)
+
+def isTopLevelAllView = my.owner == Jenkins.get();
+def canSetUpDistributedBuilds = Jenkins.get().hasPermission(Computer.CREATE) &&
+        Jenkins.get().clouds.isEmpty() &&
+        Jenkins.get().getNodes().isEmpty();
+def hasAdministerJenkinsPermission = Jenkins.get().hasPermission(Jenkins.ADMINISTER);
+def hasItemCreatePermission = my.owner.hasPermission(Item.CREATE);
+
 div {
-    h1(_("Welcome to Jenkins!"))
-
-    if (my.owner == Jenkins.get()) {
-        // we're a top-level 'All' view
-        if (Jenkins.get().hasPermission(Computer.CREATE) && Jenkins.get().clouds.isEmpty() && Jenkins.get().getNodes().isEmpty()) {
-            div(class: 'call-to-action') {
-                if (Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
-                    raw(_("distributedBuildsWithCloud"))
-                } else {
-                    raw(_("distributedBuilds"))
+
+    div(class: "empty-state-block") {
+        if (isTopLevelAllView) {
+            if (canSetUpDistributedBuilds || hasItemCreatePermission) {
+                h1(_("Welcome to Jenkins!"))
+
+                p(_("noJobDescription"))
+
+                if (canSetUpDistributedBuilds) {
+                    section(class: "empty-state-section") {
+                        h2(_("setUpDistributedBuilds"), class: "h4")
+                        ul(class: "empty-state-section-list") {
+                            li(class: "content-block") {
+                                a(href: "computer/new", class: "content-block__link") {
+                                    span(_("setUpAgent"))
+                                    span(class: "trailing-icon") {
+                                        l.svgIcon(
+                                                class: "icon-sm",
+                                                href: "${resURL}/images/material-icons/svg-sprite-navigation-symbol.svg#ic_arrow_forward_24px")
+                                    }
+                                }
+                            }
+
+                            if (hasAdministerJenkinsPermission) {
+                                li(class: "content-block") {
+                                    a(href: "computer/new", class: "content-block__link") {
```

Comment:
Hmm this doesn't seem right,

1. it's just linking to new computer, not the cloud configuration screen.
2. it's showing up when I don't have a cloud plugin installed

Although 2 probably isn't too bad as there's docs on the configure Clouds page saying what to do:

![image](https://user-images.githubusercontent.com/21194782/92975381-07ad2380-f480-11ea-8410-b98c1a20ae77.png)

```suggestion
                                    a(href: "configureClouds", class: "content-block__link") {
```

Reply:
Maybe I messed something up